realpath.sh

```diff
@@ -109,3 +109,7 @@ _gnu_stat_readlink() {
 _bsd_stat_readlink() {
     stat -f %Y -- "$1" 2>/dev/null
 }
+
+_posix_ls_readlink() {
+    LC_TIME=POSIX ls -l -- "$1" | sed -e 's/^.\{11\}\ *[0-9]\{1,\}\ *[a-zA-Z\._@\$-]\{1,\}\ *[a-zA-Z\._@\$-]\{1,\}\ *[0-9]\{1,\}\ *[^ ]*\ *[0-9]\{1,2\}\ *[^\ ]*\ \(.*\)/\1/g' -e 's/.* -> //g' 2>/dev/null
```

Comment:
Wow, it'll take me a little bit to wrap my head around that regex.

Does the `g` flag do anything when the substitution match is anchored to the beginning of the line with `^`?